### PR TITLE
Finish PlainStatusBackend

### PR DESCRIPTION
**Pull request by ralismark**
#487 implements most of this but appears abandoned. This fixes some of the comments left on those changes.

### CODE REVIEW

**Comment by pkgw** [comment]
Thanks again for your contribution! I especially appreciate your working from @Mrmaxmeier's branch to as to preserve their commits in the history.

With your update my only comment is about the test coverage. It would be nice to add something there to keep the coverage moving in the right direction. I think if you add a case or two to `test/executable.rs` that use the new `color=no` option, that should do the trick?

File: tests/executable.rs [hunk 1]
```diff
@@ -355,3 +355,33 @@ fn test_keep_logs_on_error() {
 
     assert!(log.contains(r"job aborted, no legal \end found"));
 }
+
+#[test]
+fn test_no_color() {
+    if env::var("RUNNING_COVERAGE").is_ok() {
```
[the remark this answers is not in the document]
Comment (Mrmaxmeier) [reply tests/executable.rs:361]: I vaguely remember something about the coverage tracking not working for end-to-end tests that spawn a new `tectonic` process. Not sure if that's still an issue though.

**Comment by pkgw** [comment]
I've pushed some commits that will try to re-enabled the `executable` tests in the coverage run. I'm not sure if they will correctly add to the coverage stats (as per @Mrmaxmeier's comment), but hopefully they'll at least run.

**Comment by pkgw** [comment]
OK I think it's the case that the executable tests are running, but the coverage induced by executing the `tectonic` program is not being recorded. I'll do a bit of research to if there's anything that we can do about that ... hopefully someone has tackled this problem before.

If that doesn't work out we can certainly go ahead and merge this, but I think it's worth some effort trying to solve this.

**Comment by pkgw** [comment]
xref: https://github.com/assert-rs/assert_cmd/issues/9

**Comment by pkgw** [comment]
OK, I did a little experiment. If I hack up the `executable` test harness to manually launch its `tectonic` subprocesses inside kcov, I can make it so that we do gather coverage information from those invocations. Doing so gains us about 1.3% in code coverage (~1000 lines) compared to the current baseline. I think I could wire this up in the CI/CD framework, but it would be a bit hacky and ugly.

Alternatively, we could work on revising the `executable`-type tests to use Tectonic as a library, with an absolutely minimal amount of code in `src/main.rs` (or its moral equivalent).

Or, we could keep those tests as-is and just keep in mind that they won't contribute to the code coverage stats.

I am not sure which of these options I prefer. Any opinions?

**Comment by pkgw** [comment]
OK, well, I am still interested in any thoughts about this broader question, but work along those lines is best saved for a separate PR. So, let's merge this one!

**Pull request merged**